### DISCUSSION TITLE
Searxng Latency

### DISCUSSION THREAD

**Discussion by derek8bai** [searxng/searxng#2445] [General]
Our Searxng latency is pretty good. 1 sec on average, 2 seconds for our P99 latency. But we want to make the P99 latency 1 second as well, even if it means less comprehensive results. 

Anyone have any suggestions for this?

**Comment by Klewltz** [comment]
I want low latency in searxng at low load because searching intermittently is faster and results are displayed in 1 second (i.e. at high load) on average instead of 3 or 2 at low load.

**Comment by unixfox** [comment]
You can disable some engines and just keep one.

**Reply by Klewltz** [reply]
Not are engines, is how to works the searxng because no works at hight velocity at low load and at hight load for example 99% cpu works fast

**Comment by dalf** [comment]
I'm not sure to fully understand, but with 1 second budget, some engines won't have to time to process the query: they will time out from the SearXNG point of view.

You can set the max timeout to 1 second on the server: 
https://github.com/searxng/searxng/blob/b3d29cb86db4cc1a4e6320016529d1361451e1f1/searx/settings.yml#L161

Also, there is an undocumented (?) feature: you can set the max timeout per query

For example: 
* `<1 linux` search for `linux` and set the timeout to 1 second.
* `<500 linux` search for `linux` and set the timeout at 500ms.

`<` can lower the default timeout or it can extend it. For example: `<10 !slowengine linux` search for `linux` using the engine named `slowengine` with a timeout 10 seconds (even if the timeout for engine is 3).

See https://github.com/searx/searx/pull/1640#issuecomment-517227357 for the details.

**Reply by return42** [reply]
Didn't ever know there is an `<` syntax :)
